### PR TITLE
implemented cross-parsing algorithm to estimate relative entropy

### PR DESCRIPTION
**Pull request by bmguo**

### CODE REVIEW

File: source/kkp/kkp.h [hunk 1]
```diff
@@ -60,4 +60,7 @@ int kkp2(unsigned char *X, int *SA, int n,
 int kkp1s(unsigned char *X, int n, std::string SA_fname,
     std::vector<std::pair<int, int> > *F);
 
+int parse_phrase(unsigned char* X, int n, int i, int psv, int nsv,
```
Comment (smcantab) [inline source/kkp/kkp.h:63]: where is the function definition? Did you forget to push the changes to the .cpp file?

File: source/sweetsourcod/lempel_ziv.hpp [hunk 2]
```diff
@@ -25,8 +25,8 @@ std::string int_vector_to_string(const std::vector<T>& v){
     ss.reserve(v.size());
     for (const auto& x: v){
         if (x < 0) {throw std::runtime_error("int_vector_to_string only accepts sequences of positive values");}
-        if (x > 255) {throw std::runtime_error("x>255, exceeded ascii table");}
-        ss.push_back(static_cast<int>(x));
+        if (x > 254) {throw std::runtime_error("x>255, exceeded ascii table");}
```
Comment (smcantab) [inline source/sweetsourcod/lempel_ziv.hpp]: let's write 2 versions of this function, one for 1 to 255 to be use with the cross parser and one for 0-255 to be used with the regular LZ

File: sweetsourcod/lempel_ziv.pyx [hunk 3]
```diff
@@ -20,4 +20,12 @@ def lempel_ziv_factors(lattice, version='lz77'):
             factors = get_lz77_factors(lattice)
     else:
         raise NotImplementedError
-    return factors
\ No newline at end of file
+    return factors
+    
+
+cpdef cross_parsing_complexity(lattice1, lattice2):
```
Comment (smcantab) [inline sweetsourcod/lempel_ziv.pyx:26]: why is this cpdef and the other cdef?

**Review by smcantab — CHANGES_REQUESTED** [review]

File: source/kkp/kkp.h (hunk 1 above)
Comment (bmguo) [reply source/kkp/kkp.h:63]: that's an auxiliary routine already defined in kkp.cpp

File: sweetsourcod/lempel_ziv.pyx (hunk 3 above)
Comment (bmguo) [reply sweetsourcod/lempel_ziv.pyx:26]: cross_parsing_complexity returns the complexity and the sumlog, while the other returns the direct parsing results (list of {position, length})

**Comment by smcantab** [comment]
@bmguo if you resolve the issues above I can merge

**Pull request merged**